cloudify_rest_client/cluster.py

```diff
@@ -163,14 +167,21 @@ def add(self, host_ip, node_name):
         Use the credentials returned as part of the ClusterNode structure
         when joining the cluster.
 
+        database_ip is used to validate architecture compatibility of the
+        cluster nodes
+
         :param host_ip: the externally-visible IP of the current node
         :param node_name: the name of this node used internally in the cluster
         :return: representation of the node that will join the cluster
         :rtype: ClusterNode
         """
+        response = self.api.get('/rest/config', filter='postgresql_host')
```

Comment:
it's weird that we ask the node for its postgres host, and then send it back to it, no? i'd prefer for the node to figure it out itself, because this request means that joining is now 3 requests instead of 2

Reply:
Fixed, moved validations to the joining node's side rather than the Master node's